prepublish.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+set -eo pipefail
+
+RED="\033[0;31m"
+GREEN='\033[0;32m'
+NC='\033[0m' # No color
+
+echo "${GREEN}Pulling latest from GitHub...${NC}"
+git checkout master
+git pull
+
+echo "${GREEN}Cleaning directory and fresh installing...${NC}"
+git clean -fdx
+yarn install
+
+echo "${GREEN}Building files and running tests...${NC}"
+yarn test
+yarn build
+
+echo "${GREEN}Bumping version and creating tagged release commit...${NC}"
+yarn bump
+PACKAGE_VERSION=$(node -pe "require('./package.json').version")
+BRANCH="release-$PACKAGE_VERSION"
+git checkout -b $BRANCH
+git add package.json
+git commit -m "Bump package version to $PACKAGE_VERSION"
+
+echo "${GREEN}Pushing tag and release commit to Github...${NC}"
+TAG="v$PACKAGE_VERSION"
+
+# Push up release branch containing the updated package versions
+git push --set-upstream origin $BRANCH
+git tag -a $TAG -m "Release $TAG" -s
```

Comment:
`Annotated Tags` is more secure but it does require users to set up GPG key. Not sure if you all are okay to annotate the release tags or prefer to use lightweight tags?
https://stackoverflow.com/questions/4971746/why-should-i-care-about-lightweight-vs-annotated-tags/4971817#4971817

Reply:
I'm unfamiliar with both, although lightweight seems fine to me